From 421d5cc13785214d20d40e1e8d4eb049da9a8aa9 Mon Sep 17 00:00:00 2001
From: "Liu, Kai1" <kai1.liu@intel.com>
Date: Mon, 27 Nov 2023 15:52:06 +0800
Subject: [PATCH] Initiate VehiclePropValue object when creating it

The random value of VehiclePropValue will trigger timeout when
getting property OBD2_LIVE_FRAME in getCurrentLiveFrame function
of DiagnosticHalService.

Tracked-On: OAM-113603
Signed-off-by: Liu, Kai1 <kai1.liu@intel.com>
---
 automotive/vehicle/2.0/default/common/src/VehicleUtils.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/automotive/vehicle/2.0/default/common/src/VehicleUtils.cpp b/automotive/vehicle/2.0/default/common/src/VehicleUtils.cpp
index c16b29a2e..03a9df579 100644
--- a/automotive/vehicle/2.0/default/common/src/VehicleUtils.cpp
+++ b/automotive/vehicle/2.0/default/common/src/VehicleUtils.cpp
@@ -30,7 +30,7 @@ namespace V2_0 {
 
 std::unique_ptr<VehiclePropValue> createVehiclePropValue(
     VehiclePropertyType type, size_t vecSize) {
-    auto val = std::unique_ptr<VehiclePropValue>(new VehiclePropValue);
+    auto val = std::unique_ptr<VehiclePropValue>(new VehiclePropValue());
     switch (type) {
         case VehiclePropertyType::INT32:      // fall through
         case VehiclePropertyType::INT32_VEC:  // fall through
-- 
2.34.1

